alerts/cluster-monitoring-operator/PrometheusScrapeBodySizeLimitHit.md

```diff
@@ -0,0 +1,114 @@
+# PrometheusScrapeBodySizeLimitHit
+
+## Meaning
+
+The alert `PrometheusScrapeBodySizeLimitHit` is triggered when at least one of
+Prometheus' scrape targets replies with a response body larger than the
+configured `body_size_limit`.
+By default, there is no limit on the body size of the scraped targets.
+When set, this limit prevents Prometheus from consuming excessive amounts of memory
+when scraped targets return a response that is deemed too large.
+
+## Impact
+
+Metrics coming from targets responding with a body exceeding the configured
+size limit aren't ingested by Prometheus. The targets will be considered as being
+down and they will have their `up` metric set to 0 which may also trigger the
+`TargetDown` alert.
+
+## Diagnosis
+
+We can find the value of the configured body size limit in the configmap `cluster-monitoring-config`
+in the namespace `openshift-monitoring`.
+
+To check the value in configmap, use this command:
+```bash
+oc get cm -n openshift-monitoring cluster-monitoring-config -o yaml | grep enforcedBodySizeLimit
+```
+
+To find out which targets are exceeding the body size limit, we can refer
+to the Openshift Web UI, go to the Observe > Targets page, and check which
+targets are down.
+
+If we need more information than the UI provides, such as discovered labels
+and scrape pool, we can query the Prometheus API endpoint `/api/v1/targets`.
+
+In the field `data.activeTargets` search for those that have `health` field
+different than `up` and check for the `lastError` field for confirmation:
+
+1. Get the name of the secret containing the token of service account `prometheus-k8s`.
+We use the following command and check for the name  `prometheus-k8s-token-[a-z]+`
+in the line `Tokens`. In the example below, the secret name is `prometheus-k8s-token-nwtrf`.
```

Comment:
IIRC the SA token isn't populated anymore in the secret. The admin should either create a token as the service account (`oc create token prometheus-k8s -n openshift-monitoring`) or use their own credentials. The first option is probably good enough.

Reply:
The token seems still to exist in 4.11. Here is what I got from a CI 4.11 cluster:
```bash
➜  oc describe sa prometheus-k8s -n openshift-monitoring
Name:                prometheus-k8s
Namespace:           openshift-monitoring
Labels:              app.kubernetes.io/component=prometheus
                     app.kubernetes.io/instance=k8s
                     app.kubernetes.io/name=prometheus
                     app.kubernetes.io/part-of=openshift-monitoring
                     app.kubernetes.io/version=2.36.2
Annotations:         serviceaccounts.openshift.io/oauth-redirectreference.prometheus-k8s:
                       {"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"prometheus-k8s"}}
Image pull secrets:  prometheus-k8s-dockercfg-bv5pf
Mountable secrets:   prometheus-k8s-dockercfg-bv5pf
Tokens:              prometheus-k8s-token-rs4gf
Events:              <none>
➜  oc version
Client Version: 4.10.16
Server Version: 4.11.0-0.nightly-2022-06-30-005428
Kubernetes Version: v1.24.0+9ddc8b1
```
I will add how to create the token for service account in the paragraph, too.

Reply:
added an line using this command to create token for prometheus-k8s
`oc sa new-token prometheus-k8s -n openshift-monitoring`